### ISSUE TITLE
Wasteland (1988)

### ISSUE THREAD

**Issue by Python-Exoproject** [exoscoriae/eXoDOS#1172]
Just noticed this one has three conf files simply due to having three different machine modes for graphics options. Now that we know about the -machine flag this could be cut down to one conf.

**Comment by exoscoriae** [comment]
this doesn't work, as each game still has it's own executable\folder, so just changing the machine type doesn't solve it.

**Comment by Python-Exoproject** [comment]
[eXoDOS.zip](https://github.com/exoscoriae/eXoDOS/files/4908037/eXoDOS.zip)

This will do it. Its still a non standard launch bat file, but now its just that and not that plus three different confs. Archive includes new launch bat, new dosbox.conf, and new game file structure.

**Comment by Python-Exoproject** [comment]
Bugger, just thought of a flaw in that logic. It would reset any saved game every load. Can it be tweaked or just go back to your way?

**Comment by exoscoriae** [comment]
good thinking, but I agree that wiping out the folder wouldn't be great.

If we *really* wanted to explore this route, I think the solution is a passed variable via that launch menu.

eg:
if user presses a button for "Tandy" it echos "TANDY" to a file called TANDY.SEL. Then calls the conf file with machine type set to tandy. The conf file calls a new bat file called select.bat. Select then says if exist TANDY.SEL then does "cd Tandy", WL. 

So it would have 4 different sections of the batch file, each identical to the existing conf file autoexec sections - but rather than being determined after dosbox starts, they are determined by the existence of a TANDY.SEL, CGA.SEL, COMP.SEL, or EGA.SEL file.

When you make a selection for something like Tandy, it would first del any existing .SEL files, to ensure 2 can't exist at the same time. Hell, it could even be made smart to check for a .SEL when it is first run, and if it detects the file the loader menu is modifed to something like:

Previously you played the game with Tandy, would you like to use this again or select a different game version?

And then switch to the main selection menu if they choose not to use the same one again. 

This makes it easier for someone who is actually playing through the game, so if they came back a week later - they don't have to remember which version they were working on.

How does this sound? I can script it up tomorrow morning or you can start messing with it if you want to take a crack at it.

**Comment by exoscoriae** [comment]
I went ahead and scripted it up. Runs nicely. Try picking something, exiting, and then running again.

**Comment by Python-Exoproject** [comment]
It works well. My only niggle would be that as a user now that im asked if I want to change graphics mode like that it implies thats all thats changing, but the reality is that if I change graphics mode I would have a whole new savegame and party. I suggest either:
1. tell the user as much with a warning about using another mode (I dont like that option), 
2. tweak it so that "GAME1" and "GAME2" are copied to whichever mode is chosen from whichever mode was last chosen.
3. use some of my original idea and have run.bat copy the relevant graphics mode files into an operating folder like we do for sound choices, ensuring "GAME1" & "GAME2" are always left alone in the operating folder.

**Comment by exoscoriae** [comment]
CURS, MASKS.WLF, and WLA.BIN are different between EGA and TANDY.
However, COLOR.FNT, CURS, IC0-9.WLF, MASKS.WLF. and WLA.BIN are different between EGA & CGA
Meanwhile, only WLA.BIN is different between CGA & CGA Composite.

I'm not getting into a game of knowing which exact files need to change based on what version they had versus which they are going to. That seems absolutely silly when it already works the way it is. Option 3 is off the table for this reason. It doesn't save any significant space (the whole game is less than a megabyte), it's messy, and it is really really bad for preservation. 

I'll throw a warning in the run.bat file that save games are locked to machine types. I feel like enough hand holding has already gone into this game. It already worked before, as we did all this just because it had extra conf files. Frankly, I don't think it was worth all of this time just to remove a few conf files. But I did it for the sake of it. To go back in and start checking old folders for modified game save files and move them around.... I'll never get anything done if I'm doing stuff like that.

**Comment by Python-Exoproject** [comment]
As discussed in PM I came up with a method for having universal savegame between graphics modes. I used run.bat

I also have included in the attached archive a new version of the launch bat file as one thing had been overlooked. The way it was setup if you selected to use the same mode that was previously chosen it would have graphical issues if that mode was Tandy or Composite due to the machine not being set when you selected "yes". I fixed it by adding an extra label that sets the machine variable. I also removed the comment that saves arent compatible as they are now (as long as you accept my method).

[wasteland.zip](https://github.com/exoscoriae/eXoDOS/files/4932993/wasteland.zip)
